tests/v3/test_store/test_remote.py

```diff
@@ -40,18 +41,31 @@ def get_boto3_client():
 
     # NB: we use the sync botocore client for setup
     session = Session()
-    return session.create_client("s3", endpoint_url=endpoint_uri)
+    return session.create_client("s3", endpoint_url=endpoint_url)
 
 
 @pytest.fixture(autouse=True, scope="function")
 def s3(s3_base):
+    """
+    Quoting Martin Durant:
+    pytest-asyncio creates a new event loop for each async test.
+    When an async-mode s3fs instance is made from async, it will be assigned to the loop from
+    which it is made. That means that if you use s3fs again from a subsequent test,
+    you will have the same identical instance, but be running on a different loop - which fails.
+
+    For the rest: it's very convenient to clean up the state of the store between tests,
+    make sure we start off blank each time.
+
+    https://github.com/zarr-developers/zarr-python/pull/1785#discussion_r1634856207
+    """
     client = get_boto3_client()
     client.create_bucket(Bucket=test_bucket_name, ACL="public-read")
     s3fs.S3FileSystem.clear_instance_cache()
-    s3 = s3fs.S3FileSystem(anon=False, client_kwargs={"endpoint_url": endpoint_uri})
+    s3 = s3fs.S3FileSystem(anon=False, client_kwargs={"endpoint_url": endpoint_url})
     s3.invalidate_cache()
     yield s3
-    requests.post(f"{endpoint_uri}/moto-api/reset")
+    requests.post(f"{endpoint_url}/moto-api/reset")
+    client.close()
```

Comment:
+1

Reply:
despite adding this, I'm still seeing output like this at the end of my tests:

> Task was destroyed but it is pending!
task: <Task pending name='Task-10' coro=<StoreTests.test_get() done, defined at /home/bennettd/dev/zarr-python/src/zarr/testing/store.py:83> wait_for=<Future pending cb=[Task.task_wakeup()]>>
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x7f7f53484c70>

Reply:
It's probably client in the async filesystem instance. The full workflow should look like
```
s3 = S3FileSystem(..., asynchronous=True)
session = await s3.set_session()
...
await session.close()
```

Reply:
thanks for the pointers, those warnings are now gone!